nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListFile.java

```diff
@@ -0,0 +1,378 @@
+
+        final String fullPath = fileInfo.getFullPathFileName();
+        final File file = new File(fullPath);
+        final Path filePath = file.toPath();
+        final Path directoryPath = new File(getPath(context)).toPath();
+
+        final Path relativePath = directoryPath.relativize(filePath.getParent());
+        String relativePathString = relativePath.toString() + "/";
+        if (relativePathString.isEmpty()) {
+            relativePathString = "./";
+        }
+        final Path absPath = filePath.toAbsolutePath();
+        final String absPathString = absPath.getParent().toString() + "/";
+
+        attributes.put(CoreAttributes.PATH.key(), relativePathString);
+        attributes.put(CoreAttributes.FILENAME.key(), fileInfo.getFileName());
+        attributes.put(CoreAttributes.ABSOLUTE_PATH.key(), absPathString);
+
+        try {
+            FileStore store = Files.getFileStore(filePath);
+            if (store.supportsFileAttributeView("basic")) {
+                try {
+                    final DateFormat formatter = new SimpleDateFormat(FILE_MODIFY_DATE_ATTR_FORMAT, Locale.US);
+                    BasicFileAttributeView view = Files.getFileAttributeView(filePath, BasicFileAttributeView.class);
+                    BasicFileAttributes attrs = view.readAttributes();
+                    attributes.put(FILE_SIZE_ATTRIBUTE, Long.toString(attrs.size()));
+                    attributes.put(FILE_LAST_MODIFY_TIME_ATTRIBUTE, formatter.format(new Date(attrs.lastModifiedTime().toMillis())));
+                    attributes.put(FILE_CREATION_TIME_ATTRIBUTE, formatter.format(new Date(attrs.creationTime().toMillis())));
+                    attributes.put(FILE_LAST_ACCESS_TIME_ATTRIBUTE, formatter.format(new Date(attrs.lastAccessTime().toMillis())));
+                } catch (Exception ignore) {
+                } // allow other attributes if these fail
+            }
+            if (store.supportsFileAttributeView("owner")) {
+                try {
+                    FileOwnerAttributeView view = Files.getFileAttributeView(filePath, FileOwnerAttributeView.class);
+                    attributes.put(FILE_OWNER_ATTRIBUTE, view.getOwner().getName());
+                } catch (Exception ignore) {
+                } // allow other attributes if these fail
+            }
+            if (store.supportsFileAttributeView("posix")) {
+                try {
+                    PosixFileAttributeView view = Files.getFileAttributeView(filePath, PosixFileAttributeView.class);
+                    attributes.put(FILE_PERMISSIONS_ATTRIBUTE, PosixFilePermissions.toString(view.readAttributes().permissions()));
+                    attributes.put(FILE_GROUP_ATTRIBUTE, view.readAttributes().group().getName());
+                } catch (Exception ignore) {
+                } // allow other attributes if these fail
+            }
+        } catch (IOException ioe) {
+            // well then this FlowFile gets none of these attributes
```

Comment:
This is the same logic that is in GetFile.  It is basically a graceful decay depending on what features are supported on the system, the catch is necessary because even though it has checked that the view is supported with "supportsXXXAttributeView" the calls to the "getXXXAttributeView" methods declare the exception.  I agree with the decision not to log it since most platforms support at least one of the views.  In environments that don't support any views the name and path will still be returned, and logging an exception on those systems will log for every file processed.